rwilliams659/overlook-app
# Rw manager dash right side

#### What changes/additions are a part of this PR?
This PR adds significant functionality to the right side of the manager dashboard. This involved:
- Adding a method to my booking class that is able to sort bookings by date (most recent to oldest) using Moment.js
- Adding a querySelector/event listener for a click on the "Search" button on the search bar
- Adding a function so that when an invalid name is entered, an error message displays
- Adding functions to that when a valid name is entered, a box is loaded that displays the user's name, total spent on rooms, and a list of bookings from most recent to oldest. Only bookings after today's date (which is currently none as I have "today" set as the real "today") have a "Delete reservation" button that appears next to it. 

#### Any background context you want to provide?
I struggled with implementing Moment.js, but I think I was able to get it working now. I'd love any feedback on UX that you might have, or any bugs in functionality you have. Also open to any refactoring suggestions, as I feel like some of my code is a big of a maize!

I  also have not yet hooked up the "Delete reservation" or "Add reservation" buttons to do DELETE/POST requests so those don't function at this time. 

#### Where should the reviewer start?
Most code has been added to index.js. HTML & CSS changes are relatively minor and not really necessary to review. 

#### How should this be tested (test suite/manually)?
I don't yet have a test suite for index.js. You can run the BookingRepo test to test the new sortBookingsByDate function. Most testing will need to be done by testing out the app. When you get to the login screen, enter manager as the username & overlook2020 as the password, and you should see the manager dashboard, so you can test it out!

#### What are the relevant tickets?
Ticket #36 

#### Any questions/additional comments:
Thanks for your help! P.S. I know I'm doing iteration 4 before finishing 2 & 3...oops!
